File: 백준/Silver/9881. Ski Course Design/Ski Course Design.py
N = int(input())
hills = list(int(input()) for _ in range(N))
answer = 0

min_v = min(hills)
max_v = max(hills)

def cal(min, max):
    result = 0
    for i in range(N):
        tmp = 0
        if hills[i] > max:
            tmp = (hills[i] - max) ** 2
        elif hills[i] < min:
            tmp = (min - hills[i]) ** 2
        result += tmp
    return result

answer = 1e9
for i in range(min_v, max_v - 17):
    answer = min(answer, cal(i, i+17))
if answer == 1e9:
    print(0)
else:
    print(answer)

File: 백준/Silver/9881. Ski Course Design/Ski Course Design.java
import java.util.*;
import java.io.*;

public class Main {
    static int N;
    static int[] hills;
    public static void main(String[] args) throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        N = Integer.parseInt(br.readLine());
        hills = new int[N];
        int max = -1;
        int min = 101;
        for (int i = 0; i < N; i++) {
            hills[i] = Integer.parseInt(br.readLine());
            if (hills[i] > max) {
                max = hills[i];
            }

            if (hills[i] < min) {
                min = hills[i];
            }
        }

        int answer = Integer.MAX_VALUE;
        for (int tmp = min; tmp <= max-17; tmp++) {
            int result = 0;
            for (int i = 0; i < N; i++) {
                if (hills[i] < tmp) {
                    result += Math.pow((tmp - hills[i]), 2);
                } else if (hills[i] > tmp + 17) {
                    result += Math.pow((hills[i] - (tmp + 17)), 2);
                }
            }
//            System.out.println("tmp : " + tmp + " result : " + result);
            if (answer > result) {
                answer = result;
            }
        }
        if (answer == Integer.MAX_VALUE) {
            System.out.println(0);
        } else {
            System.out.println(answer);
        }



    }
}
